### ISSUE TITLE
Taxii2 connector failing because of missing parameters?

### ISSUE THREAD

**Issue by SamWoolerton** [OpenCTI-Platform/connectors#1128]
## Description

I'm configuring a Taxii2 connector (the generic one in this repo) to access a private threat feed.
The connector is reaching the data source and sending data to OpenCTI, but it doesn't appear in OpenCTI anywhere.
I've checked the logs from the OpenCTI worker containers and they're throwing this error
```
2023-05-08T02:23:18.130415324Z INFO:root:Message processed, thread terminated
2023-05-08T02:23:18.130538126Z INFO:root:Message (delivery_tag=154) acknowledged
2023-05-08T02:23:18.132078251Z INFO:root:Processing a new message (delivery_tag=155), launching a thread...
2023-05-08T02:23:18.133041967Z INFO:pycti.api:Importing a indicator (id: indicator--8af2a857-8bea-4bf8-87a0-a94a89551da7)
2023-05-08T02:23:18.133439974Z ERROR:pycti.entities:[opencti_indicator] Missing parameters: name or pattern or pattern_type or x_opencti_main_observable_type
```

I've checked the response via Postman directly and it looks fine (though I'm not super familiar with STIX/TAXII2 so could be missing something).
Indicators all have name and pattern, there are just some marking definition entries at the start that don't have these fields. I assumed these would be handled by OpenCTI though, so would surprise me if they're what's causing it.

OpenCTI now has a label _benign_ that it didn't before (that's in the source data), so it's clearly pulled some data in but not the indicators themselves.

## Environment

1. OS (where OpenCTI server runs): Dockerised (run from Docker Compose via Portainer)
2. OpenCTI version: { e.g. OpenCTI 1.0.2 }: using image opencti/platform:5.6.2
3. OpenCTI client: { e.g. frontend or python }: frontend
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Add the taxii2 connector for this feed and configure with environment variables
2. Run it (works perfectly)
3. Error in the OpenCTI import workers

Environment variables (interpolated with ${} via the Portainer UI).
Note I tried both versions below of connector type and scope separately, and get the same response regardless.

```
  connector-taxii2:
    image: opencti/connector-taxii2:5.6.2
    environment:
      - OPENCTI_URL=http://opencti:8080
      - OPENCTI_TOKEN=${OPENCTI_ADMIN_TOKEN}
      - CONNECTOR_ID=${CONNECTOR_IMPORT_TAXII_ID} # Valid UUIDv4
      # - CONNECTOR_TYPE=EXTERNAL_IMPORT
      - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
      - CONNECTOR_NAME=TAXII2CONNECTOR
      - CONNECTOR_SCOPE=application/json
      # - CONNECTOR_SCOPE=ipv4-addr,ipv6-addr,vulnerability,domain,url,file-sha256,file-md5,file-sha1
      - CONNECTOR_CONFIDENCE_LEVEL=65 # From 0 (Unknown) to 100 (Fully trusted)
      - CONNECTOR_UPDATE_EXISTING_DATA=false
      - CONNECTOR_LOG_LEVEL=info
      - CONNECTOR_AUTO=true # Enable/disable auto-import of file
      - CONNECTOR_VALIDATE_BEFORE_IMPORT=true # Validate any bundle before import
      - CONNECTOR_ONLY_CONTEXTUAL=false # Only extract data related to an entity (a report, a threat actor, etc.)
      - TAXII2_DISCOVERY_URL=REDACTED # Required (test endpoint)
      - TAXII2_USERNAME=${TAXII2_USERNAME} # Required
      - TAXII2_PASSWORD=${TAXII2_PASSWORD}
      - TAXII2_V21=false # This is using v2.0
      - TAXII2_COLLECTIONS=api.* # Required
      - TAXII2_INITIAL_HISTORY=48 # Required, in hours
      - TAXII2_INTERVAL=1 # Required, in hours
      - TAXII2_VERIFY_SSL=true
      - TAXII2_CREATE_INDICATORS=true # Generate indicators for ingested observables
      - TAXII2_CREATE_OBSERVABLES=true # Generate observables for ingested indicators
    restart: always
    depends_on:
      - opencti
```

## Expected Output

Expect it to import fine

**Comment by annoyingapt** [comment]
I think this issue is related to Taxii v2.0 servers not including pattern_type, which I fixed in 5.7.0. Try the latest version, see if it works. https://github.com/OpenCTI-Platform/connectors/commit/685c230624457b3b78410448642a5018222ca483 Also, there is a bug in the latest version, which will cause it to bail if the Taxii server has no manifest. I just created a PR for this, so would suggest keeping an eye on this if you want to wait for a 100% working new version. https://github.com/OpenCTI-Platform/connectors/pull/1130
